js/Edit.jsx

```diff
@@ -0,0 +1,330 @@
+// get hljs from highlight library
+let hljs = require('./highlight.js');
+
+// Initialize the worker
+let worker = null;
+
+// Initialize the debug
+let debug = false;
+if ('1' === highlight_obj.debug) {
+	debug = true;
+}
+
+// Initialize required components
+const {Component} = wp.element;
+const {SelectControl} = wp.components;
+const {__} = wp.i18n;
+
+// Edit class for edit method of registeredBlockType
+class Edit extends Component {
+
+	// Constructor
+	constructor() {
+
+		super(...arguments);
+
+		// set area height of textarea
+		let temp_txtarea = document.createElement('textarea');
+		temp_txtarea.value = this.props.attributes.content;
+		document.body.appendChild(temp_txtarea);
+
+
+		// Init states
+		this.state = {
+			updateMessage : false,
+			status        : '',
+			html_content  : '',
+			content       : this.props.attributes.content,
+			contentChanged: true,
+			language      : this.props.attributes.language,
```

Comment:
I'll have to explain the login for this. I necessarily have to change the state because of other state attributes. If I update attributes also, render method will be called 2 times for every character change. To prevent this I used states only, and attributes will be changed only when the user wants to see the preview (when the block loses focus).